ml-agents/mlagents/trainers/saver/saver.py

```diff
@@ -0,0 +1,39 @@
+# # Unity ML-Agents Toolkit
+import abc
+from mlagents.trainers.policy import Policy
+
+
+class BaseSaver(abc.ABC):
```

Comment:
I think "BaseSerializer" is a better name for this since it both saves and loads (I know the TF class is called Saver, but it has a bad name).

Reply:
Hmm but we use the term "Serializing" for model exporting. "Checkpoint" could be a better word here. I chose Saver over Checkpointer just because it sounds better.

Reply:
"serialize" generally means converting between object instances and persistent formats. This also loads, so I disagree that Saver sounds better.

Reply:
By better I mean "saver" and "checkpointer". The problem is, the term "serialize" has been saved for the use of exporting model to onnx/barracuda by us (model_serialization.py does **tf graph -> frozen_graph.pb -> .nn/.onnx**). Here the saver is more referring to checkpointint tf (**save/load .ckpt files**) and calls external export function for exporting when needed. So I would separate these two and not using the term serialize for the whole saver.

Or I can use serialize here and modify model_serialization.py to other names like model_export.py

Reply:
OK, I'm fine with the name as it is.